### ISSUE TITLE
Map render breaks whenever I scroll in or out/ Importing chunks does not work across worlds

### ISSUE THREAD

**Issue by pmeyourpillow** [Querz/mcaselector#22]
Okay, so whenever I zoom in or out of the map, the (cache?) breaks, and what I mean is that the regions/chunks unload and go grey. The only way I could fix this is by clearing the cache everytime this happens. At the same time, whenever I try to export chunks across worlds, it does not work, either by doing nothing at all or by scattering parts of the chunks in random places.

**Comment by Querz** [comment]
Looks like the "Overwrite existing chunks" setting does not work properly, i'll have a look at that.

As for the render issues: Does this happen immediately after you started the program or only after it has been running for a while? Is it happening on a specific zoom level (when you zoom far out or far in)? When you enable debug mode in the settings, could you send me the console output for the moment when these render issues happen? Could you send some screenshots?

The map view is creating differently sized images for 3 zoom levels: 512x512, 256x256 and 128x128. Whenever you zoom in or out, it is supposed to unload and load the appropriate images, that might create a "flickering" effect.

**Comment by pmeyourpillow** [comment]
It seems to only happen after the program has been running for a while, can't say for sure how long though. It happens in general, but seems to be more common when it happens at outer levels. 
![mcscreenshot](https://user-images.githubusercontent.com/52432017/60493491-c311df80-9c61-11e9-8c68-c9939bfbe5ad.png) This is the only screenshot I could snag since it usually greys out

**Comment by Querz** [comment]
That's weird.
What Java version are you using to run the program?
What OS are you running on?

I managed to fix the bug with importing chunks, the overwrite setting didn't work when not interacting with the checkbox at least once. I'll create a release once we figure out what causes the map renderer to freak out.

**Comment by pmeyourpillow** [comment]
I am using Java version 8 on Windows 10.

So is the only solution to importing chunks to mess with the checkbox for now?

**Comment by Querz** [comment]
Simply deselecting and selecting the checkbox should work. Alternatively you can build the project yourself from the master branch (which has the fix for the checkbox) using gradle with the following tasks:
`build minifyCss shadowJar`

**Comment by Querz** [comment]
@pmeyourpillow The most recent release (1.7.3) fixes the issue with the checkbox.

**Comment by beskhu3epnm** [comment]
I'm experiencing this bug as well. Although mine might be a bit different. I load in the world, when I zoom in or out, it unloads the cache, and the only way to correct is to reopen the region folder. I have not tried to export a world yet with deleted chunks. If you reopen the region zoomed in, the same happens for zooming out far enough. Attached is a video illustrating the bug: https://streamable.com/ghj1o

I'm using Linux Mint 19 with the latest version of mcaselector.

**Comment by Boidushya** [comment]
@bekshu3epnm Im expereiencing the same exact problem from Windows 10 Pro

**Comment by Querz** [comment]
I added a logging system in [release 1.9.4](https://github.com/Querz/mcaselector/releases/tag/1.9.4) to better address issues like this. If you enable the option `Print debug messages` in `File-->Settings`, it now creates a file called `debug.log`. This file contains debug information of the current session. In order to see if there is an error happening, can you do the following?

* Run `mcaselector-1.9.4.jar`
* Enable the `Print debug messages` option in the settings
* Recreate the issue
* Close the program (do not run MCASelector again, it will overwrite the old file!)
* Send me the `debug.log` file

Thank you all!

**Comment by beskhu3epnm** [comment]
OK I opened a world in mcaselector-1.9.4, zoomed in to a region, then zoomed out (the region disappeared), and zoomed in again (the region remains disappeared). The debug.log file can be viewed [here](https://files.catbox.moe/e9p36q.log).

**Comment by Querz** [comment]
Thanks, can you also list the contents of the `/home/user/cache/4b9ff85e5c5b3e379ababc2c587b92f3/` folder and its subfolders?

**Comment by beskhu3epnm** [comment]
There is no folder named Cache in the home directory, but a hidden Cache folder. So the directory would be `/home/user/.cache`. Inside the .Cache folder, there does not appear to be a folder called `4b9ff85e5c5b3e379ababc2c587b92f3`. I launched mcaselector again to see if it would generate a folder in .Cache, but it didn't.

**Comment by Querz** [comment]
MCASelector currently creates a cache folder in the working directory.

**Comment by Boidushya** [comment]
I don't know how, but release 1.9.4 fixed it for me! I was using 1.9.3 before. Thanks a ton!

**Comment by Querz** [comment]
@Boidushya that is good to hear, even though I did not change anything related to the map view in the 1.9.4 update.

@beskhu3epnm The `cache` folder should be at the same location as the `debug.log` file.

**Comment by Querz** [comment]
I did some testing in a Ubuntu VM using Java 8. The problem is not permission related, because if it does not have enough permissions to create the cache folder or the cache files, it is simply going to render the images again using the mca files. That's definitely not the best outcome for this situation, I'll probably add a warning dialog or something that explains this if the cache directory can't be created.

I'm also going to make a change so the cache directory will always be created in the same folder as the `mcaselector.jar` file instead of the working directory.

While going through some code I discovered a few relevant places where an exception is not printed to the `debug.log` file

**Comment by Querz** [comment]
Does this still happen or was this fixed after introducing a new caching system in 1.9.5?

**Comment by Querz** [comment]
I will close this Issue as this problem doesn't seem to occur anymore.
If people should still encounter a similar issue, please open a [new Issue](https://github.com/Querz/mcaselector/issues/new?assignees=&labels=&template=bug_report.md&title=).
